druntime/src/core/stdc/stdatomic.d

```diff
@@ -0,0 +1,161 @@
+/**
+ * A D implementation of the C stdatomic.h header.
+ *
+ * $(NOTE If it compiles it should produce similar assembly to the system C toolchain and should not introduce when optimizing unnecessary behaviors.)
+ *
+ * Copyright: Copyright Richard (Rikki) Andrew cattermole 2023.
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:   Richard (Rikki) Andrew cattermole
+ * Source:    $(DRUNTIMESRC core/stdc/stdatomic.d)
+ */
+module core.stdc.stdatomic;
+import core.atomic;
+import core.internal.atomic;
+import core.stdc.config;
+
+@safe nothrow @nogc:
+
+///
+enum memory_order
+{
+    /// No ordering provided
+    memory_order_relaxed = MemoryOrder.raw,
+    /// As per cppreference.com circa 2015 no compiler supports consume memory order and in practice it devolves to acquire.
+    memory_order_consume = MemoryOrder.acq,
+    /// Prevent reordering before operation
+    memory_order_acquire = MemoryOrder.acq,
+    /// Prevent reordering after operation
+    memory_order_release = MemoryOrder.rel,
+    /// Prevent reordering before and after operation
+    memory_order_acq_rel = MemoryOrder.acq_rel,
+    /// Prevent reordering before for read operations and after for writes.
+    memory_order_seq_cst = MemoryOrder.seq
+}
+
+///
+enum
+{
+    ///
+    ATOMIC_BOOL_LOCK_FREE = IsAtomicLockFree!bool ? 2 : 0,
+    ///
+    ATOMIC_CHAR_LOCK_FREE = IsAtomicLockFree!char ? 2 : 0,
+    ///
+    ATOMIC_CHAR16_T_LOCK_FREE = IsAtomicLockFree!wchar ? 2 : 0,
+    ///
+    ATOMIC_CHAR32_T_LOCK_FREE = IsAtomicLockFree!dchar ? 2 : 0,
+    ///
+    ATOMIC_WCHAR_T_LOCK_FREE = ATOMIC_CHAR16_T_LOCK_FREE,
+    ///
+    ATOMIC_SHORT_LOCK_FREE = IsAtomicLockFree!short ? 2 : 0,
+    ///
+    ATOMIC_INT_LOCK_FREE = IsAtomicLockFree!int ? 2 : 0,
+    ///
+    ATOMIC_LONG_LOCK_FREE = IsAtomicLockFree!c_long ? 2 : 0,
+    ///
+    ATOMIC_LLONG_LOCK_FREE = IsAtomicLockFree!ulong ? 2 : 0,
+    ///
+    ATOMIC_POINTER_LOCK_FREE = IsAtomicLockFree!(void*) ? 2 : 0,
+    ///
+    ATOMIC_CHAR8_T_LOCK_FREE = ATOMIC_CHAR_LOCK_FREE,
+}
+
+/// Guaranteed to be a atomic boolean type
+struct atomic_flag
+{
+    private bool b;
+}
+
+///
+void atomic_flag_clear()(atomic_flag* obj)
+{
+    pragma(inline, true);
```

Comment:
I see you made this function a template, probably to help with forced inlining. In my experience it is more reliable to place pragma(inline, true) _outside_ the function, such that it is always seen by semantic analysis (i.e. without inspecting the function body).

Reply:
That wasn't why I did it. Although that is a good reason to consider it.

I did it because this isn't something you want to be linking against and therefore certainly should work with -betterC.

I'm happy to change it.